python/tank/util/shotgun/connection.py

```diff
@@ -200,11 +202,54 @@ def _raise_missing_key(key):
         # Make sure it is None.
         config_data["app_store_http_proxy"] = None
 
+    config_data["host"] = cleanup_host(config_data["host"])
+
     return config_data
 
 
+def cleanup_host(server_url):
```

Comment:
i feel the methodname could be more descriptive: it's not just cleaning up the host name, it's cleaning up protocol, slashes etc. `cleanup_url` or `cleanup_url_hostname`?